project/src/main/java/com/google/sps/data/DataAccessor.java

```diff
@@ -70,7 +72,26 @@ public boolean isRegistered(String userId) {
   public void registerUser(String userId) {
     checkArgument(!Strings.isNullOrEmpty(userId), "Invalid user ID");
     checkArgument(!isRegistered(userId), "User already registered.");
-    Entity userEntity = new Entity(USER_ENTITY_NAME, userId);
+    Entity userEntity = new Entity(USER_ENTITY_KIND, userId);
     datastoreService.put(userEntity);
   }
+
+  /**
+   * Stores the UserPreferences in the personalized user database.
+   *
+   * @param userId The ID of the user to store the preferred cuisines for.
+   * @param userPref The user choices on the query form to store in the user’s database.
+   */
+  public void storeUserPreferences(String userId, UserPreferences userPref) {
+    checkArgument(!Strings.isNullOrEmpty(userId), "Invalid user ID");
+    if (userPref.cuisines().isEmpty()) {
+      return;
+    }
+    Entity prefsEntity = new Entity(PREFERNCES_ENTITY_KIND);
```

Comment:
Can't this whole thing be in-line?

datastoreService.put(new Entity(PREFERNCES_ENTITY_KIND).setProperty(..).setProperty(..));

Reply:
This is not possible because setProperty returns void, so it all has to be set seperately.